pypeit/spectrographs/mdm_modspec.py

```diff
@@ -0,0 +1,317 @@
+"""
+Module for MDM/Modspec specific methods.
+
+.. include:: ../include/links.rst
+"""
+
+import array as arr
+import numpy as np
+
+from astropy.time import Time
+
+from pypeit import msgs
+from pypeit import telescopes
+from pypeit.core import framematch
+from pypeit.spectrographs import spectrograph
+from pypeit.core import parse
+from pypeit.images import detector_container
+
+class MDMModspecEchelleSpectrograph(spectrograph.Spectrograph):
+    """
+    Child to handle MDM Modspec Echelle instrument+detector
+    """
+    ndet = 1
+    name = 'mdm_modspec_echelle'
+
+    telescope = telescopes.HiltnerTelescopePar()
+
+    camera = 'Echelle'
+    header_name = 'Modspec'
+    pypeline = 'MultiSlit'
+    supported = True
+    comment = 'MDM Modspec spectrometer'
+
+    pypeline='MultiSlit'
+
+    def get_detector_par(self, det, hdu=None):
+        """
+        Return metadata for the selected detector.
+
+        Args:
+            det (:obj:`int`):
+                1-indexed detector number.
+            hdu (`astropy.io.fits.HDUList`_, optional):
+                The open fits file with the raw image of interest.  If not
+                provided, frame-dependent parameters are set to a default.
+
+        Returns:
+            :class:`~pypeit.images.detector_container.DetectorContainer`:
+            Object with the detector metadata.
+        """
+        # Detector 1
+        # See Echelle at 2.4m f/7.5 scale : http://mdm.kpno.noirlab.edu/mdm-ccds.html
+        gain = np.atleast_1d([1.3])      # Hardcoded in the header 
+        ronoise = np.atleast_1d([7.90])    # Hardcoded in the header
+        lenSpat = hdu[0].header['NAXIS1']     # length of spatial axis, including overscan. Horizontal axis of original .fits files
+        lenSpec = hdu[0].header['NAXIS2']      # length of spectral axis. Vertical axis of original .fits files
+    
+        datasec = np.atleast_1d([
+            '[{0:d}:{1:d},{2:d}:{3:d}]'.format(1, lenSpec, 1, 300)
+        ])
+        oscansec = np.atleast_1d([
+            '[{0:d}:{1:d},{2:d}:{3:d}]'.format(1, lenSpec, 308, lenSpat)
+
+        ])
+        if hdu is None:
```

Comment:
Since the functionality exists, I wonder if the binning should call component_meta:

`binning = self.compound_meta(self.get_headarr(hdu), "binning")`